### DISCUSSION TITLE
Loading multiple kitty sessions under MacOS

### DISCUSSION THREAD

**Discussion by steliyan** [kovidgoyal/kitty#6606] [Q&A]
I am using MacOS and try to move away from using tmux altogether (kinda inspired by [this comment](https://github.com/kovidgoyal/kitty/issues/391#issuecomment-638320745)), but also, I don't really care that much about the persistent session, but more about having a pre-defined layouts for different projects I am working on.

Of course, a natural solution seemed to use the [--session](https://sw.kovidgoyal.net/kitty/invocation/#cmdoption-kitty-session) flag. This works great for having a nice default, but falls apart if you have a very project specific setup.

How? Well, if I have a running kitty instance and I do: `kitty --session my_session_file`, I then get a new OS window, but I also get a new kitty icon in the MacOS Dock (undesirable). Having `--single-instance` to the `macos-launch-services-cmdline` file does _NOT_ change the behavior.

I've tried using `open -a kitty.app --args --session my_session_file` ([ref](https://github.com/kovidgoyal/kitty/issues/4690)), but it's not creating the specified session. If a kitty instance is already running, it just focuses the OS window.

So I guess my question boils down to the following:

> How can I load a session in a new OS kitty window keeping the window grouping in MacOS?

Am I missing something or should I leverage kitty's remote control capabilities?

**Comment by kovidgoyal** [comment]
Use remote control for this, or configure macos' dock to group all kitty
instances together (I dont use mac so not sure if the latter is doable).

**Comment by kovidgoyal** [comment]
Oh and also you can combine --session and --single-instance, then your session will open in new os windows in the existing instance

**Reply by steliyan** [reply]
This works very weird, IMHO. Not sure if it's a bug or a feature, tho. :))

I'll try to test a bit more, however using a `--single instance` (`-1`) seems to work OK, but there is not need to "detach" (to add `&` or `& disown` or `open -a`) which is unexpected for me.

**Reply by kovidgoyal** [reply]
Why would it be unexpected. When you use -1 it means kitty is sending a message to the existing instance to open new windows. Once the message is sent and acknowledged it exits. How else would it possibly work?
